### PR TITLE
create Matrix class

### PR DESCRIPTION
**Pull request by egorgur**
It overrides __add__() and __str__() methods.
__add()__ returns new Matrix with the same number of rows and columns and with the summed values.

### CODE REVIEW

File: Matrix.py
Comment (mrFroxyk) [inline Matrix.py:1]: Always write ths dockstring

File: Matrix.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Matrix:
+    def __init__(self, *args, **kwargs):
+        self.matrix_list = list(args)
+
+    def __str__(self):
+        return str(self.matrix_list)
+
+    def __add__(self, other):
```
Comment (mrFroxyk) [inline Matrix.py]: Please, write the anotation for the `other` arg

File: main.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from Matrix import Matrix
+
+if __name__ == '__main__':
+    m1 = Matrix([1, 2, 3], [2, 3, 4])
+    m2 = Matrix([1, 2, 3], [2, 3, 4])
```
Comment (mrFroxyk) [inline main.py]: Don`t use a short names (like a `m1`)

File: Matrix.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Matrix:
+    def __init__(self, *args, **kwargs):
+        self.matrix_list = list(args)
```
Comment (mrFroxyk) [inline Matrix.py]: It is prefereble if the `__init__` method accepts only self and inint_data with anotation
Like a 
```suggestion
    def __init__(self: Self, init_data: list[list[int | float]]):
        self.matrix_list = init_data
```

Comment (mrFroxyk) [reply Matrix.py]: If you want, you can separate an annotation at the top of the file.
```python
Row = list[int | float]
MatrixType = list[Row]

class Matrix:
    def __init__(self: Self, init_data: MatrixType ):
        self.matrix_list = init_data
```

File: poetry.lock
Comment (mrFroxyk) [inline poetry.lock:1]: There cannot be a `poetry.lock` file here without `pyproject.toml`

**Review by mrFroxyk — CHANGES_REQUESTED** [review]
I meant you should named the branch like a `#feature/POLY-9-implement-matrix-class`, not a `#feature/POLY-9-summary-description`
[summary description](https://www.deepl.com/ru/translator#en/ru/summary%20description)

File: poetry.lock
Comment (egorgur) [reply poetry.lock:1]: pyproject.toml is already in the repository

Comment (mrFroxyk) [reply poetry.lock:1]: Why poetry.lock not in the master?  You can 
1 add a poetry.lock in the master branch
2 rebase the matrix branch onto master
3 delete poetry.lock from the matrix branch

The .lock file should only change if the pyproject has changed

Comment (egorgur) [reply poetry.lock:1]: there are only two branches. The main and the feature/poly-9... at the moment. The pyproject.toml exists in the both branches

File: Matrix.py (hunk 3 above)
Comment (egorgur) [reply Matrix.py]: If __init__ method takes only one argument, then it is not possible to implement the required functionality.
Poly-9 task suggests that the new Matrix object takes infinite number of arguments which are lists(rows) of the matrix

Comment (mrFroxyk) [reply Matrix.py]: I am eblan)0)0)))
You are right
I actualized task. It was
```python
x = Matrix([1, 1], [2, 2])
```

now
```python
x = Matrix([[1, 1], [2, 2]])
```

Comment (mrFroxyk) [reply Matrix.py]: Thank you for your attention

File: poetry.lock
Comment (mrFroxyk) [reply poetry.lock:1]: Yes you're right
But the PR should include only the necessary changes. The `.lock` file for this branch is not  necessary change

File: matrix.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    Represents the mathematical matrix and implements the method of the sum of two matrix.
+
+    """
+
+    def __init__(self: Self, init_data: list[list[int | float]] | tuple[list[int | float], ...]):
+        """
+        Init method.
+
+        Takes the matrix parameters (width and height of a matrix) as a list of lists.
+        """
```
Comment (mrFroxyk) [inline matrix.py]:
```suggestion
        """Init method. """
```
It's better to write a simply docksting if there aren't need a special explanations to a hard buisness logic

Comment (mrFroxyk) [reply matrix.py]: Fix in all code

File: matrix.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+class Matrix:
+    """
+    Matrix class.
+
+    Represents the mathematical matrix and implements the method of the sum of two matrix.
+
+    """
+
+    def __init__(self: Self, init_data: list[list[int | float]] | tuple[list[int | float], ...]):
```
Comment (mrFroxyk) [inline matrix.py]:
```suggestion
    def __init__(self: Self, init_data: Sequence[Sequence[int | float]]:
```
You can use `Sequence` from typing

File: poetry.lock
Comment (egorgur) [reply poetry.lock:1]: Alright, I get it. I will not do that mistake

**Review by mrFroxyk — CHANGES_REQUESTED** [review]

File: matrix.py (hunk 4 above)
Comment (egorgur) [reply matrix.py]: I fixed. See the latest changes

File: matrix.py (hunk 5 above)
Comment (egorgur) [reply matrix.py]: Done

File: matrix.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    Matrix class.
+
+    Represents the mathematical matrix and implements the method of the sum of two matrix.
+
+    """
+
+    def __init__(self: Self, init_data: Sequence[Sequence[int | float]]):
+        """Init method."""
+        print(init_data)
```
Comment (mrFroxyk) [inline matrix.py]:
```suggestion
```

File: matrix.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        self.matrix_list = init_data
+
+    def __str__(self: Self):
+        """Str method."""
+        return str(self.matrix_list)
+
+    def __add__(self: Self, other: Self):
+        """Summ method."""
```
Comment (mrFroxyk) [inline matrix.py]:
```suggestion
        """Sum method."""
```

**Review by mrFroxyk — APPROVED** [review]
[Typing SVG](https://git.io/typing-svg)

**Pull request merged**